packages/dev/loaders/src/SPLAT/splatFileLoader.ts

```diff
@@ -45,6 +47,147 @@ export class SPLATFileLoader implements ISceneLoaderPluginAsync, ISceneLoaderPlu
         return false;
     }
 
+    /**
+     * Code from https://github.com/dylanebert/gsplat.js/blob/main/src/loaders/PLYLoader.ts
+     * Loads a .ply from data array buffer
+     * if data array buffer is not ply, returns the original buffer
+     */
+    private _loadPLY(data: any): ArrayBuffer {
+        const ubuf = new Uint8Array(data);
+        const header = new TextDecoder().decode(ubuf.slice(0, 1024 * 10));
+        const headerEnd = "end_header\n";
+        const headerEndIndex = header.indexOf(headerEnd);
+        if (headerEndIndex < 0 || !header) {
+            return data;
+        }
+        const vertexCount = parseInt(/element vertex (\d+)\n/.exec(header)![1]);        
+
+        let rowOffset = 0;
+        const offsets: Record<string, number> = {
+            double: 8,
+            int: 4,
+            uint: 4,
+            float: 4,
+            short: 2,
+            ushort: 2,
+            uchar: 1,
+        };
+
+        type PlyProperty = {
+            name: string;
+            type: string;
+            offset: number;
+        };
+        const properties: PlyProperty[] = [];
+        for (const prop of header
+            .slice(0, headerEndIndex)
+            .split("\n")
+            .filter((k) => k.startsWith("property "))) {

```

Comment:
would be better to cache this before, because unless you change the `header` array, 3 new arrays will be created on each iteration

Reply:
header is coming from the .ply and is different from 1 to the other. it's not possible to cache it.

Reply:
Is it not possible to do this:

```suggestion
        const filtered = header
            .slice(0, headerEndIndex)
            .split("\n")
            .filter((k) => k.startsWith("property ")
        for (const prop of filtered)) {

```

Reply:
yes, committed